sdlib/d/gc/tcache.d

```diff
@@ -388,36 +432,69 @@ unittest getCapacity {
 	void* tlPtr = &threadCache;
 	assert(threadCache.getCapacity(tlPtr[0 .. 100]) == 0);
 
-	// Check capacity for an appendable GC allocation.
-	auto p0 = threadCache.allocAppendable(100, false);
-
-	// p0 is appendable and has the minimum large size.
-	// Capacity of segment from p0, length 100 is 16384:
-	assert(threadCache.getCapacity(p0[0 .. 100]) == 16384);
-	assert(threadCache.getCapacity(p0[1 .. 100]) == 16383);
-	assert(threadCache.getCapacity(p0[50 .. 100]) == 16334);
-	assert(threadCache.getCapacity(p0[99 .. 100]) == 16285);
-	assert(threadCache.getCapacity(p0[100 .. 100]) == 16284);
```

Comment:
The test change so that the large size are tested using large size (and don't trip the slab behavior in that diff) shoudl go on their own.

It is trivial for a reviewer to check that the new test cover the same cases, and it is also trivial to see that this part of the test is unaffected in this diff. On the other end, as things stands, it's almost impossible to know anything about what'(s going on with testing without involved detective work.

Reply:
Then maybe the alloc should have been 16000 byes in length or so and voila. That's your first change. Then we can easily see that the behavior for large alloc has not changed, except the test for the max behavior. Instead we have a see of changes.